uiTests/pages/kimchi_project/login.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import os
+import utils
+
+from selenium.common.exceptions import TimeoutException
+
+# locators by ID
+USERNAME = "username"
+PASSWORD = "password"
+LOGIN_BUTTON = "btn-login"
+LOGIN_BAR = "user-login"
+
+# environment variables
+ENV_USER = "KIMCHI_USERNAME"
+ENV_PASS = "KIMCHI_PASSWORD"
+ENV_PORT = "KIMCHI_PORT"
+ENV_HOST = "KIMCHI_HOST"
+
+
+class KimchiLoginPage():
+    """
+    Page object to Login
+
+    Expect environment variables:
+    KIMCHI_USERNAME: username for the host
+    KIMCHI_PASSWORD: password for the host
+    KIMCHI_HOST: host for kimchi
+    KIMCHI_PORT: port for kimchi
+    """
+
+    def __init__(self, browser):
+        self.browser = browser
+
+        # assert envs
+        assert ENV_USER in os.environ, f"{ENV_USER} is a required environment var"
+        assert ENV_PASS in os.environ, f"{ENV_PASS} is a required environment var"
+        assert ENV_HOST in os.environ, f"{ENV_HOST} is a required environment var"
+
+        # get values
+        self.host = os.environ[ENV_HOST]
+        self.port = os.environ.get(ENV_PORT) or "8001"
+        self.user = os.environ[ENV_USER]
+        self.password = os.environ[ENV_PASS]
+
```

Comment:
I am a little bit confused. It says I need to have KIMCHI_* vars and the test uses a different set of variables.
How do they glue together?

Reply:
no, the test uses `KIMCHI_`. In the snippet above, Im just using constants to save the strings